.github/workflows/docker.yml

```diff
@@ -22,6 +22,18 @@ jobs:
       contents: read
       packages: write
     steps:
+      - name: Checkout code
+        uses: actions/checkout@v5
+        with:
+          persist-credentials: false
+
+      - name: Lint Dockerfile
+        uses: hadolint/hadolint-action@v3
+        with:
+          dockerfile: Dockerfile
+          ignore: DL3007,DL3008,DL3013,DL3016
+          failure-threshold: warning
+
```

Comment:
Can you make this its own workflow instead, that runs on pull requests to master and dev? This docker.yml only runs on a schedule and merges to master/dev only, and I feel like linting then is too late.